commands/diagnostic.go

```diff
@@ -0,0 +1,131 @@
+	manet "github.com/multiformats/go-multiaddr-net"
+	sysi "github.com/whyrusleeping/go-sysinfo"
+)
+
+// prior art: https://github.com/ipfs/go-ipfs/blob/c1bc202f41ab07debdecd806ba97cec0a1241bf7/core/commands/sysdiag.go
+
+var diagCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Show info about the filecoin node",
+	},
+	Subcommands: map[string]*cmds.Command{
+		"sys": sysDiagCmd,
+	},
+}
+
+// SysDiagResult type returned from systems diagnostics command.
+type SysDiagResult struct {
+	Disk    map[string]interface{} `json:"disk"`
+	Env     map[string]interface{} `json:"environment"`
+	Mem     map[string]interface{} `json:"memory"`
+	Net     map[string]interface{} `json:"network"`
+	Runtime map[string]interface{} `json:"runtime"`
+}
+
+var sysDiagCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Print system diagnostic information.",
+		ShortDescription: `
+Prints out information about your computer to aid in easier debugging.
+`,
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		details := SysDiagResult{
+			Disk:    make(map[string]interface{}),
+			Env:     make(map[string]interface{}),
+			Mem:     make(map[string]interface{}),
+			Net:     make(map[string]interface{}),
+			Runtime: make(map[string]interface{}),
+		}
+		if err := sysInfo(details); err != nil {
+			return err
+		}
+
+		return cmds.EmitOnce(res, details)
+	},
+}
+
+func sysInfo(details SysDiagResult) error {
+	runtimeInfo(details.Runtime)
+	envVarInfo(details.Env)
+	if err := diskSpaceInfo(details.Disk); err != nil {
+		return err
+	}
+	if err := memInfo(details.Mem); err != nil {
+		return err
+	}
+	if err := netInfo(details.Net); err != nil {
+		return err
+	}
+	return nil
+}
+
+func runtimeInfo(out map[string]interface{}) {
+	out["os"] = runtime.GOOS
+	out["arch"] = runtime.GOARCH
+	out["numcpu"] = runtime.NumCPU()
+	out["version"] = runtime.Version()
+	out["compiler"] = runtime.Compiler
+	out["gomaxprocs"] = runtime.GOMAXPROCS(0)
+	out["numgoroutines"] = runtime.NumGoroutine()
+}
+
+func envVarInfo(out map[string]interface{}) {
+	out["GOPATH"] = os.Getenv("GOPATH")
+	out["FIL_API"] = os.Getenv("FIL_API")
+	out["FIL_PATH"] = os.Getenv("FIL_PATH")
+}
+
+func diskSpaceInfo(out map[string]interface{}) error {
+	dinfo, err := sysi.DiskUsage(filPath())
+	if err != nil {
+		return err
+	}
+	out["fstype"] = dinfo.FsType
+	out["free_space"] = dinfo.Free
+	out["total_space"] = dinfo.Total
+	return nil
+}
+
+func memInfo(out map[string]interface{}) error {
+	meminf, err := sysi.MemoryInfo()
+	if err != nil {
+		return err
+	}
+
+	out["swap"] = meminf.Swap
+	out["virt"] = meminf.Used
+	return nil
+}
+
+func netInfo(out map[string]interface{}) error {
+	addrs, err := manet.InterfaceMultiaddrs()
+	if err != nil {
+		return err
+	}
+
+	var straddrs []string
+	for _, a := range addrs {
+		straddrs = append(straddrs, a.String())
+	}
+
+	out["interface_addresses"] = straddrs
+	return nil
+}
+
+func filPath() string {
```

Comment:
Blocking: This won't work if a user specified home directory with the `--repodir` flag, you should parse that arg, import `repo` and use `repo.GetRepoDir(repoDirArgString)` function as in other commands.  Also this way the env var stuff is taken care of for you.